src/libraries/Microsoft.Extensions.Hosting/src/HostBuilder.cs

```diff
@@ -127,13 +129,40 @@ public IHost Build()
             }
             _hostBuilt = true;
 
+            // REVIEW: If we want to raise more events outside of these calls then we will need to
+            // stash this in a field.
+            using var diagnosticListener = new DiagnosticListener("Microsoft.Extensions.Hosting");
```

Comment:
I know we have the tests in the HostFactoryResolver, but it might be a good idea to add some tests directly to Extensions.Hosting that ensure this fires correctly. To prevent breaking it in the future.

Reply:
Done